### PR TITLE
Do not accept invalid units #982

### PR DESCRIPTION
**Pull request by abdelr**
Do not accept invalid units #982

### CODE REVIEW

**Review by Yuri05 — APPROVED** [review]

File: src/OSPSuite.Infrastructure.Import/Core/DataFormat/MixColumnsDataFormat.cs [hunk 1]
```diff
@@ -39,6 +44,7 @@ protected override UnitDescription ExtractUnits(string description, IUnformatted
                .Substring(1, units.Length - 2) //remove the brackets
                .Trim()                         //remove whitespace
                .Split(',')                     //split comma separated list
+               .Where(unitName => _dimensionFactory.Dimensions.SelectMany(dimension => dimension.Units).Any(unitFromDimension => unitFromDimension.Name == unitName)) //only accepts valid units
```
Comment (msevestre) [inline src/OSPSuite.Infrastructure.Import/Core/DataFormat/MixColumnsDataFormat.cs:47]: I do not think this is correct here. We have a lot of methods checking whether a unit is valid or not (taking care of case, synonyms etc...) 
This needs to change

**Review by msevestre — CHANGES_REQUESTED** [review]
Checking of units is not what I expected. Look at existing method 

For instance
IDimension DimensionForUnit(string unitName)

if the retruend value is null., it is not a valid dimension.

File: src/OSPSuite.Infrastructure.Import/Core/DataFormat/MixColumnsDataFormat.cs [hunk 2]
```diff
@@ -39,6 +44,7 @@ protected override UnitDescription ExtractUnits(string description, IUnformatted
                .Substring(1, units.Length - 2) //remove the brackets
                .Trim()                         //remove whitespace
                .Split(',')                     //split comma separated list
+               .Where(unitName => _dimensionFactory.DimensionForUnit(unitName) != null) //only accepts valid units
```
Comment (msevestre) [inline src/OSPSuite.Infrastructure.Import/Core/DataFormat/MixColumnsDataFormat.cs:47]: The same logic should be used for other unit recognition by the way (in column headeR)
Is this the case @abdelr and @georgeDaskalakis

**Review by msevestre — COMMENTED** [review]
merging. Please review my comment

**Pull request merged**